roles/models.py

```diff
@@ -0,0 +1,37 @@
+"""
+General models for the micromasters app
+"""
+
+from django.contrib.auth.models import User
+from django.db import models
+
+from courses.models import Program
+
+
+class Role(models.Model):
+    """
+    Assignment of a MicroMasters Role per program to a user.
+
+    The role names MUST have a correspondent logical role in roles.py
+    and the name MUST be a snake-case version of the correspondent class name
+    Example: if the logical role class in roles.py is `FooRole`, the
+    instance here must me `foo_role`.
+    """
+    # roles names
+    STAFF = 'staff'
+    INSTRUCTOR = 'instructor'
+
+    ALL_ROLES_CHOICES = [
+        (STAFF, STAFF),
+        (INSTRUCTOR, INSTRUCTOR),
+    ]
+    user = models.ForeignKey(User)
+    program = models.ForeignKey(Program)
+    role = models.CharField(
+        choices=ALL_ROLES_CHOICES,
+        default=STAFF,
+        max_length=30,
+    )
```

Comment:
(continuing the conversation from the main thread where I linked to the gist...)

that example from django documentation is a scenario where the choices are simple strings, and there aren't actual classes associated with the choices. we have specific classes associated with these choices, and the choices should correspond to them directly.

how about this instead:

```python
# roles/roles.py
class Staff(AbstractUserRole):
    # ...
    role_id = 'staff'

class Instructor(AbstractUserRole):
    # ...
    role_id = 'instructor'

# roles/models.py
class Role(models.Model):
    # ...
    ASSIGNABLE_ROLES = [Staff, Instructor]
    DEFAULT_ROLE = Staff
    # ...
    role = models.CharField(
        choices=[(role.role_id, role.role_id) for role in ASSIGNABLE_ROLES],
        default=DEFAULT_ROLE.role_id,
        max_length=30,
    )

```

similar to what i posted with a couple key differences. there is a formal relationship between the model and the actual Role classes - `ASSIGNABLE_ROLES` is defined directly on the model, it references Roles directly, and builds the choices from that list. i just think it's very awkward to have the role choice name ('staff', etc) defined outside of the actual Role class when it's basically Role class metadata. any new Role we create should have an associated choice available for the model, and i don't see a good reason why we should need to define data for that Role in two different places. added benefit: the above example doesn't require a comment in the docstring to guide the creation of these choices